nisystemlink/clients/testmonitor/models/_query_products_request.py

```diff
@@ -0,0 +1,90 @@
+from enum import Enum
+from typing import List, Optional
+
+from nisystemlink.clients.core._uplink._json_model import JsonModel
+
+
+class ProductField(Enum):
+    """The valid ways to order a product query."""
+
+    ID = "ID"
+    FAMILY = "FAMILY"
+    PART_NUMBER = "PART_NUMBER"
+    NAME = "NAME"
+    UPDATED_AT = "UPDATED_AT"
+
+
+class QueryProductsBase(JsonModel):
+    filter: Optional[str] = None
+    """
+    The product query filter in Dynamic Linq format.
+
+    Allowed properties in the filter are:
+    - `id`: String for the global identifier of the product
+    - `partNumber`: String representing the part number of the product
+    - `name`: String of the product name
+    - `family`: String for the product family
+    - `updatedAt`: ISO-8601 formatted UTC timestamp indicating when the product was last updated.
+    - `keywords`: A list of keyword strings
+    - `properties`: A dictionary of additional string to string properties
+    - `fileIds`: A list of string ids for files stored in the file service (`/nifile`)
+
+    See [Dynamic Linq](https://github.com/ni/systemlink-OpenAPI-documents/wiki/Dynamic-Linq-Query-Language)
+    documentation for more details.
+
+    `"@0"`, `"@1"` etc. can be used in conjunction with the `substitutions` parameter to keep this
+    query string more simple and reusable.
+    """
+
+    substitutions: Optional[List[str]] = None
+    """String substitutions into the `filter`.
+
+    Makes substitutions in the query filter expression. Substitutions for the query expression are
+    indicated by non-negative integers that are prefixed with the "at" symbol. Each substitution in
+    the given expression will be replaced by the element at the corresponding index (zero-based) in
+    this list. For example, "@0" in the filter expression will be replaced with the element at the
+    zeroth index of the substitutions list.
+    """
+
+
+class QueryProductsRequest(QueryProductsBase):
+
+    order_by: Optional[ProductField] = None
```

Comment:
It is not robust to rely on the camel case convertor of `JsonModel` to convert `order_by` into `orderBy`, instead explicitly define a `Field` alias.

```suggestion
    order_by: Optional[ProductField] = Field(None, alias="orderBy")
```

Reply:
Thanks. Updated.